### PR TITLE
Add assertions and logging

### PR DESCRIPTION
**Pull request by vibes-863**

### CODE REVIEW

File: src/main/java/budgetbuddy/BudgetBuddy.java [hunk 1]
```diff
@@ -31,23 +39,50 @@ public class BudgetBuddy {
...
         DataStorage dataStorage = new DataStorage();
         this.accountManager = dataStorage.loadAccounts();
         this.transactions = dataStorage.loadTransactions(accountManager.getExistingAccountNumbers());
     }
 
     /**
-     * Main entry-point for the java.BudgetBuddy application.
+     * Sets up the logger for the BudgetBuddy application.
      */
+    private static void setupLogger() {
```
Comment (ShyamKrishna33) [inline src/main/java/budgetbuddy/BudgetBuddy.java:58]: Nice work with logging!

**Review by ShyamKrishna33 — APPROVED** [review]
Overall LGTM

File: src/main/java/budgetbuddy/BudgetBuddy.java [hunk 2]
```diff
@@ -31,23 +39,50 @@ public class BudgetBuddy {
...
+            File logsDir = new File("logs");
+            if (!logsDir.exists()) {
+                logsDir.mkdir();
+            }
+            FileHandler fh = new FileHandler("logs/budgetBuggyLog.log");
+            fh.setFormatter(new SimpleFormatter());
+            fh.setLevel(Level.INFO);
+            logger.addHandler(fh);
+        } catch (IOException e) {
+            UserInterface.printLoggerSetupError();
```
Comment (Vavinan) [inline src/main/java/budgetbuddy/BudgetBuddy.java:71]: Good job on catching exception

**Review by Vavinan — APPROVED** [review]
Overall looks good!!

**Pull request merged**